File: esque-core/src/main/java/org/loesak/esque/core/elasticsearch/documents/MigrationRecord.java
package org.loesak.esque.core.elasticsearch.documents;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonTypeName;

import java.time.Instant;
import java.util.Objects;

@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
@JsonTypeName(value = "migration")
public record MigrationRecord(
        String migrationKey,
        Integer order,
        String filename,
        String version,
        String description,
        Integer checksum,
        String installedBy, // user if any
        Instant installedOn,
        Long executionTime) {

    public MigrationRecord {
        Objects.requireNonNull(migrationKey);
        Objects.requireNonNull(order);
        Objects.requireNonNull(filename);
        Objects.requireNonNull(version);
        Objects.requireNonNull(description);
        Objects.requireNonNull(checksum);
        Objects.requireNonNull(installedOn);
        Objects.requireNonNull(executionTime);
    }

}

File: esque-core/src/main/kotlin/org/loesak/esque/core/elasticsearch/documents/MigrationRecord.kt
package org.loesak.esque.core.elasticsearch.documents

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.annotation.JsonTypeName
import java.time.Instant

@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
@JsonTypeName(value = "migration")
data class MigrationRecord(
    @get:JvmName("migrationKey") @JsonProperty("migrationKey") val migrationKey: String,
    @get:JvmName("order") @JsonProperty("order") val order: Int,
    @get:JvmName("filename") @JsonProperty("filename") val filename: String,
    @get:JvmName("version") @JsonProperty("version") val version: String,
    @get:JvmName("description") @JsonProperty("description") val description: String,
    @get:JvmName("checksum") @JsonProperty("checksum") val checksum: Int,
    @get:JvmName("installedBy") @JsonProperty("installedBy") val installedBy: String?,
    @get:JvmName("installedOn") @JsonProperty("installedOn") val installedOn: Instant,
    @get:JvmName("executionTime") @JsonProperty("executionTime") val executionTime: Long,
)
